cpp/logger/SFAwsLogger.cpp

```diff
@@ -21,6 +21,33 @@ namespace
     // Find sub string in given string
     return data.find(toSearch, pos);
   }
+
+  struct awslog_mutex
```

Comment:
why do we have two structs for awslog_mutex and awssdk_mutex, These definitions look pretty common to me.

Reply:
> why do we have two structs for awslog_mutex and awssdk_mutex, These definitions look pretty common to me and should be deduplicated

Fixed to use common definition.